tests/plugins_t/test_auditor.py

```diff
@@ -0,0 +1,118 @@
+from tardis.plugins.auditor import Auditor
+from tardis.utilities.attributedict import AttributeDict
+from tardis.interfaces.siteadapter import ResourceStatus
+from tardis.resources.dronestates import AvailableState, DownState
+
+from datetime import datetime
+from unittest import TestCase
+from unittest.mock import patch
+
+from ..utilities.utilities import async_return
+from ..utilities.utilities import run_async
+
+
+class TestAuditor(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.mock_config_patcher = patch("tardis.plugins.auditor.Configuration")
+        cls.mock_auditorclientbuilder_patcher = patch(
+            "tardis.plugins.auditor.pyauditor.AuditorClientBuilder"
+        )
+
+        cls.mock_config = cls.mock_config_patcher.start()
+        cls.mock_auditorclientbuilder = cls.mock_auditorclientbuilder_patcher.start()
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.mock_config_patcher.stop()
+        cls.mock_auditorclientbuilder_patcher.stop()
+
+    def setUp(self):
+        self.address = "127.0.0.1"
+        self.port = 8000
+        self.timeout = 20
+        self.user = "user-1"
+        self.group = "group-1"
+        self.site = "testsite"
+        self.cores = 12
+        self.memory = 100
+        self.drone_uuid = "test-drone"
+        self.machine_type = "test_machine_type"
+        config = self.mock_config.return_value
+        config.Plugins.Auditor.host = self.address
+        config.Plugins.Auditor.port = self.port
+        config.Plugins.Auditor.timeout = self.timeout
+        config.Plugins.Auditor.user = self.user
+        config.Plugins.Auditor.group = self.group
+        config.Plugins.Auditor.components.test_machine_type = AttributeDict(
+            Cores=AttributeDict(HEPSPEC=1.2, BENCHMARK=3.0),
+            Memory=AttributeDict(BLUBB=1.4),
+        )
+        config.Sites = [AttributeDict(name=self.site)]
+        config.testsite.MachineTypes = [self.machine_type]
+        config.testsite.MachineMetaData = AttributeDict(
+            test_machine_type=AttributeDict(Cores=self.cores, Memory=self.memory)
+        )
+
+        self.test_param = AttributeDict(
+            site_name=self.site,
+            machine_type=self.machine_type,
+            created=datetime.now(),
+            updated=datetime.now(),
+            resource_status=ResourceStatus.Booting,
+            drone_uuid=self.drone_uuid,
+        )
+
+        builder = self.mock_auditorclientbuilder.return_value
+        builder = builder.address.return_value
+        builder = builder.timeout.return_value
+        self.client = builder.build.return_value
+        self.client.add.return_value = async_return()
+        self.client.update.return_value = async_return()
+
+        self.plugin = Auditor()
+
+    def test_notify(self):
```

Comment:
Please, add a test for a different state than `AvailableState` and `DownState` and check that it does nothing.

Reply:
Regarding the latter point: Could you help me with the test, please? If I do not set `config.Plugins.Auditor.{user, group}` in `setUp(..)`, then those two fields will be `MagicMock` objects which breaks everything. I am unsure how to indicate `undef` (`None` obviously doesn't work either).

Reply:
Sure. I would replace https://github.com/MatterMiners/tardis/blob/445d4227b3af8b38ac445a5fe5916f72b3844d0b/tests/plugins_t/test_auditor.py#L41-L50 by something similar to 

```python
config.Plugins.Auditor = AttributeDict(host=self.address, 
                                                              ...)
```

Reply:
You need to make `config` and instance variable `self.config` and then you can simply delete single keys. 

```python
del self.config.Plugins.Auditor.user
```

Reply:
Thanks a lot!! I had to create a new plugin instance in the test in order to "reload" the changed config.